cns/restserver/api.go

```diff
@@ -1029,7 +1031,27 @@ func (service *HTTPRestService) unpublishNetworkContainer(w http.ResponseWriter,
 
 	ctx := r.Context()
 
-	if !service.isNetworkJoined(req.NetworkID) {
+	var unpublishBody nmagent.DeleteContainerRequest
+	var azrNC bool
+	err = json.Unmarshal(req.DeleteNetworkContainerRequestBody, &unpublishBody)
+	if err != nil {
+		// If the body contains only `""\n`, it is non-AZR NC
+		// In this case, we should not return an error
+		// However, if the body is not `""\n`, it is invalid and therefore, we must return an error
+		if !bytes.Equal(req.DeleteNetworkContainerRequestBody, []byte(`""`+"\n")) {
+			http.Error(w, fmt.Sprintf("could not unmarshal delete network container body: %v", err), http.StatusBadRequest)
+			return
+		}
+	} else {
+		// If unmarshalling was successful, it is an AZR NC
+		azrNC = true
```

Comment:
This is too much knowledge about NMAgent quirks in a component that shouldn't care. I'd recommend making a `nmagent.NewDeleteContainerRequest` that returns a request and an error. Then all the API needs to care about is whether or not that function says it was able to create a request or not. Then, ideally... you can just ask the request if it's an AZR NC or not instead of needing the other bool for it.

Reply:
We are not really creating any request. Just passing along the body that CNS received to NMA as payload.